app/scripts/metamask-controller.actions.test.js

```diff
@@ -185,7 +210,19 @@ describe('MetaMaskController', function () {
       const result1 = metamaskController.keyringController.state;
       await metamaskController.createNewVaultAndRestore('test@123', TEST_SEED);
       const result2 = metamaskController.keyringController.state;
-      expect(result1).toStrictEqual(result2);
+
+      // on restore, a new keyring metadata is generated
+      expect(result2).toStrictEqual(
```

Comment:
Nit: But just to make this even more explicit:

```suggestion
      // On restore, a new keyring metadata is generated.
      expect(result1.keyringsMetadata.id).toBe(mockULIDs[0]);
      expect(result2).toStrictEqual(
```